src/sentry/plugins/interfaces/releasehook.py

```diff
@@ -107,22 +124,31 @@ def set_commits(self, version, commit_list):
 
                 ReleaseCommit.objects.create(
                     organization_id=project.organization_id,
-                    project_id=project.id,
                     release=release,
                     commit=commit,
                     order=idx,
                 )
 
     def finish_release(self, version, **values):
         values.setdefault('date_released', timezone.now())
-        values.setdefault('organization', self.project.organization_id)
-        release, created = Release.objects.create_or_update(
-            version=version,
-            project=self.project,
-            values=values
-        )
-        if created:
-            release.add_project(self.project)
+        with transaction.atomic():
+            release_qs = Release.objects.filter(
```

Comment:
This pattern needs to be refactored to avoid unnecessary queries. You have like potentially, 5 queries here.

I think something like this would be the same:

```python
affected = Release.objects.filter(
  version=version,
  organization_id=self.project.organization_id,
  projects=self.project,
).update(**values)
if not affected:
    affected = Release.objects.filter(
        version=version,
        organization_id=self.project.organization_id,
    ).update(**values)
    if not affected:
        release = Release.objects.create(
            version=version,
            organization_id=self.project.organization_id,
            **values
        )
        release.add_project(self.project)
```

This would end up in best-case, 1 query just to UPDATE. Worst case would be 4, for the two SELECTS and 2 INSERTs.